server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -171,199 +162,105 @@ static void clearLoadedModules()
+   */
+  public static File[] getExtensionFilesToLoad(ExtensionsConfig config)
   {
-    URLClassLoader loader = loadersMap.get(coordinate);
-    if (loader == null) {
-      final CollectRequest collectRequest = new CollectRequest();
-
-      DefaultArtifact versionedArtifact;
-      try {
-        // this will throw an exception if no version is specified
-        versionedArtifact = new DefaultArtifact(coordinate);
-      }
-      catch (IllegalArgumentException e) {
-        // try appending the default version so we can specify artifacts without versions
-        if (defaultVersion != null) {
-          versionedArtifact = new DefaultArtifact(coordinate + ":" + defaultVersion);
-        } else {
-          throw e;
-        }
-      }
-
-      collectRequest.setRoot(new Dependency(versionedArtifact, JavaScopes.RUNTIME));
-      DependencyRequest dependencyRequest = new DependencyRequest(
-          collectRequest,
-          DependencyFilterUtils.andFilter(
-              DependencyFilterUtils.classpathFilter(JavaScopes.RUNTIME),
-              new DependencyFilter()
-              {
-                @Override
-                public boolean accept(DependencyNode node, List<DependencyNode> parents)
-                {
-                  if (accept(node.getArtifact())) {
-                    return false;
-                  }
-
-                  for (DependencyNode parent : parents) {
-                    if (accept(parent.getArtifact())) {
-                      return false;
-                    }
-                  }
-
-                  return true;
-                }
-
-                private boolean accept(final Artifact artifact)
-                {
-                  return exclusions.contains(artifact.getGroupId());
-                }
-              }
-          )
-      );
-
-      try {
-        final List<Artifact> artifacts = aether.resolveArtifacts(dependencyRequest);
-
-        List<URL> urls = Lists.newArrayListWithExpectedSize(artifacts.size());
-        for (Artifact artifact : artifacts) {
-          if (!exclusions.contains(artifact.getGroupId())) {
-            urls.add(artifact.getFile().toURI().toURL());
-          } else {
-            log.debug("Skipped Artifact[%s]", artifact);
-          }
-        }
-
-        for (URL url : urls) {
-          log.info("Added URL[%s]", url);
+    File[] extensionsToLoad;
+    final File rootExtensionsDir = new File(config.getExtensionsDir());
+    final List<String> toLoad = config.getToLoad();
+    if (toLoad == null) {
+      extensionsToLoad = rootExtensionsDir.exists() ? rootExtensionsDir.listFiles() : new File[]{};
+    } else {
+      int i = 0;
+      extensionsToLoad = new File[toLoad.size()];
+      for (final String extensionName : toLoad) {
+        final File extensionDir = new File(rootExtensionsDir, extensionName);
+        if (!extensionDir.exists()) {
+          throw new RuntimeException(
```

Comment:
Yes, and also IOException is a checked exception that is supposed to be somewhat handled. Since we are going to fail directly in this case, it makes more sense to use a RuntimeException or its subclass here.